src/app/models/event.model.spec.ts

```diff
@@ -97,14 +95,99 @@ describe('Unit Test :: Events Model', () => {
       oldValue: 'b',
       timestamp: '2018-05-27T08:54:44.63509Z',
       modifierId: '20694424-0841-4d6c-bfb5-bbbb0391b8db',
-      newValueRelationships: null,
-      oldValueRelationships: null,
+      newValueRelationships: [],
+      oldValueRelationships: [],
       type: null
     };
     expect(expectedOutPut).toEqual(output);
   });
 
   it('should correctly convert to ui model - 2', () => {
+    const map = new EventMapper();
+    const input: EventService = {
+      'attributes': {
+        'name': 'system.assignees',
+        'timestamp': '2018-05-27T08:54:44.63509Z'
+      },
+      'id': '5e7adc1d-2c8a-440d-9466-aedf4b2d60c2',
+      'relationships': {
+        'modifier': {
+          'data': {
+            'id': '20694424-0841-4d6c-bfb5-bbbb0391b8db',
+            'links': {
+              'related': 'https://api.prod-preview.openshift.io/api/users/20694424-0841-4d6c-bfb5-bbbb0391b8db'
+            },
+            'type': 'identities'
+          }
+        }
+      },
+      'type': 'events'
+    };
+    const output: EventUI = map.toUIModel(input);
+    const expectedOutPut: EventUI = {
+      name: 'system.assignees',
+      newValue: null,
+      oldValue: null,
+      timestamp: '2018-05-27T08:54:44.63509Z',
+      modifierId: '20694424-0841-4d6c-bfb5-bbbb0391b8db',
+      newValueRelationships: [],
+      oldValueRelationships: [],
+      type: null
+    };
+    expect(expectedOutPut).toEqual(output);
+  });
+
+  it('should correctly conver to ui model -3', () => {
+    const map = new EventMapper();
+    const input: EventService = {
+      'attributes': {
+        'name': 'system.assignees',
+        'newValue': null,
+        'oldValue': null,
```

Comment:
These will no longer be present if the `name` refers to a relational field.

Reply:
This is intentional. This test checks that the correct UI model is converted from a relational field with `null` values in attributes.

Reply:
Sure but the keys (`newValue` and `oldValue`) will not be present (not event set to `null`) in the response if the change happened in a relational response. That's what I tried to say.

Reply:
Btw. do you have tests for list values in `data.attributes.newValue` and `data.attributes.oldValue`? I couldn't find them anywhere. @vikram-raj ?

Reply:
>Sure but the keys (newValue and oldValue) will not be present (not event set to null) in the response if the change happened in a relational response. That's what I tried to say.

@kwk this is what you are looking for https://github.com/fabric8-ui/fabric8-planner/pull/2751/files/0e1f088d933b9691e111d941a404073c89b1134b#diff-54a28a37b7f272fea6972c13962b87eaR232.
I have added this test to ensure the current implementation(mapper function) doesn't break regardless attributes values are not present(or set to `null`) in response.

Reply:
> Btw. do you have tests for list values in data.attributes.newValue and data.attributes.oldValue? I couldn't find them anywhere. @vikram-raj ?

No, We dont have tests for list values `data.attributes.newValue` and `data.attributes.oldValue` because we dont have such field.

Reply:
@sahil143 you should add those tests because those fields will come. We currently support them in the backend as well.

Reply:
It is never a good thing to implement only against what we have but what we support.

Reply:
agreed!

Reply:
@kwk added tests and support for list values in `data.attributes` https://github.com/fabric8-ui/fabric8-planner/pull/2751/commits/d59bcc19b31e3e1fd2bb520c3b5ace3c574b2a2d